kedro-datasets/kedro_datasets/pandas/sql_dataset.py

```diff
@@ -498,11 +501,14 @@ def __init__(  # noqa: PLR0913
             self.adapt_mssql_date_params()
 
     @classmethod
-    def create_connection(cls, connection_str: str, connection_args: dict) -> None:
+    def create_connection(
+        cls, connection_str: str, connection_args: dict = None
```

Comment:
```suggestion
        cls, connection_str: str, connection_args: Optional[dict] = None
```

Reply:
Note that the linter has automatically replaced the `Optional` use.

Reply:
that's fine, we actually preferred this I forgot this syntax is available already.